ixmp4/core/optimization/scalar.py

```diff
@@ -0,0 +1,172 @@
+from datetime import datetime
+from typing import ClassVar, Iterable
+
+import pandas as pd
+
+from ixmp4.core.base import BaseFacade, BaseModelFacade
+from ixmp4.data.abstract import Docs as DocsModel
+from ixmp4.data.abstract import Run, Unit
+from ixmp4.data.abstract import Scalar as ScalarModel
+
+
+class Scalar(BaseModelFacade):
+    _model: ScalarModel
+    _run: Run
+    _unit: Unit
+    NotFound: ClassVar = ScalarModel.NotFound
+    NotUnique: ClassVar = ScalarModel.NotUnique
+
+    def __init__(
+        self,
+        name: str,
+        _run: Run,
+        value: float | None = None,
+        unit_name: str | None = None,
```

Comment:
Wondering if it might make sense to take a unit object here directly instead of just the name?
Effectively while loading data you will have the unit model somewhere already anyway...

Reply:
I was wondering the same thing, but @danielhuppmann said three weeks ago:
> On the unit-question, if I understand it correctly, being able to pass the unit as a string in the create/update methods would be much more user-friendly...

And I figure, if you have the `unit` object, it's not that much of a difference whether you pass in `unit` itself or `unit.name`.

Reply:
When adding an IAMC datapoint, the dataframe must have a column "unit" which is actually the string. I'd like to follow that easy-to-use approach here and have a required argument "unit" which can take a Unit object or a name as string or None for dimensionless.

Reply:
This is done, though I'll leave a note that we should provide logging information to users forgetting to set `unit_or_name` that we will assume the Scalar to be dimensionless.